sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -749,11 +749,15 @@ private void SetNameFieldsIfNull()
             bool async = true,
             CancellationToken cancellationToken = default)
         {
-            var pageRange = new HttpRange(
-                range.Offset + startOffset,
-                range.Length.HasValue ?
-                    range.Length.Value - startOffset :
-                    (long?)null);
+            HttpRange? pageRange = null;
+            if (!(range == default))
+            {
+                pageRange = new HttpRange(
+                    range.Offset + startOffset,
+                    range.Length.HasValue ?
+                        range.Length.Value - startOffset :
+                        (long?)null);
+            }
```

Comment:
Does ShareFileClient have the same issue?  https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs#L1794

Reply:
It does, but only for files that users explicitly create with a size of 0.

I attempted to address the issue, but it is breaking FileClient.Download() with unreliable stream for reasons I wasn't able to determine.

I created https://github.com/Azure/azure-sdk-for-net/issues/10898 to track.